packages/mobile/src/screens/search-screen-v2/SearchCategoriesAndFilters.tsx

```diff
@@ -1,16 +1,108 @@
+import { useCallback, useState } from 'react'
+
 import { ScrollView } from 'react-native'
 
-import { Flex, SelectablePill } from '@audius/harmony-native'
+import { Button, Flex, IconPlus, SelectablePill } from '@audius/harmony-native'
+import { useNavigation } from 'app/hooks/useNavigation'
+
+// TODO:
+// - Need to update filters to use FilterButton when the component is created
+// - Need to sort the filters to put filters with an active value first
+
+type SearchCategory = 'all' | 'tracks' | 'profiles' | 'albums' | 'playlists'
+type CategoryFilter =
+  | 'genre'
+  | 'mood'
+  | 'key'
+  | 'bpm'
+  | 'isVerified'
+  | 'isPremium'
+  | 'hasDownloads'
+
+const filterInfoMap: Record<CategoryFilter, { label: string; screen: string }> =
+  {
+    genre: { label: 'Genre', screen: 'SearchGenre' },
+    mood: { label: 'Mood', screen: 'SearchMood' },
+    key: { label: 'Key', screen: 'SearchKey' },
+    bpm: { label: 'BPM', screen: 'SearchBpm' },
+    isVerified: { label: 'Verified', screen: '' },
+    isPremium: { label: 'Premium', screen: '' },
+    hasDownloads: { label: 'Downloadable', screen: '' }
+  }
+
+const categoryFilters: Record<SearchCategory, CategoryFilter[]> = {
```

Comment:
I usually like to name this like `filtersByCategory` so then it's obvious when you do `filtersByCategory['genre']` for example